language-service/language_service/controller/__init__.py

```diff
@@ -0,0 +1,116 @@
+import os
+import traceback
+
+from flask import request
+from flask_restful import Resource
+
+from service.nlp import tag
+from service.vocab import get_definitions
+
+AUTH_TOKEN = os.getenv("AUTH_TOKEN")
+SUPPORTED_LANGUAGES = ["CHINESE", "ENGLISH", "SPANISH"]
+
+
+def is_authorized(request):
+    return (
+        "Authorization" in request.headers
+        and request.headers["Authorization"] == AUTH_TOKEN
+    )
+
+
+class DefinitionController(Resource):
+    def get(self, language=None, word=None):
+        if not is_authorized:
+            return {"error": "No authorization provided"}, 401
+
+        if language is None or language == "":
+            return {"error": "Language is required"}, 400
+
+        language = language.upper()
+        if language not in SUPPORTED_LANGUAGES:
+            return {"error": "Language %s is not supported" % language}, 400
+
+        if word is None or word == "":
+            return {"error": "Word is required"}, 400
+
+        try:
+            return (
+                [
+                    definition.to_json()
+                    for definition in get_definitions(language, word)
+                ],
+                200,
+            )
+        except Exception as e:
+            print("Error getting definition in %s for: %s" % (language, word))
+            stacktrace = traceback.format_exc()
+            print(stacktrace)
+            return {"error": "An error occurred"}, 500
+
+
+class DefinitionMultipleController(Resource):
+    def post(self, language=None):
+        if not is_authorized:
+            return {"error": "No authorization provided"}, 401
+
+        request_json = request.get_json()
+
+        if language is None or language == "":
+            return {"error": "Language is required"}, 400
+
+        language = language.upper()
+        if language not in SUPPORTED_LANGUAGES:
+            return {"error": "Language %s is not supported" % language}, 400
+
+        if request_json is None or "words" not in request_json:
+            return {"error": "Words is required"}, 400
+
+        words = request_json["words"]
+
+        try:
+            definitions = {
+                word: [
+                    definition.to_json()
+                    for definition in get_definitions(language, word)
+                ]
+                for word in words
+            }
+            return definitions, 200
+        except Exception as e:
+            print("Error getting definitions in %s for words: %s" % (language, text))
+            stacktrace = traceback.format_exc()
+            print(stacktrace)
+            return {"error": "An error occurred"}, 500
+
+
+class DocumentController(Resource):
+    def post(self, language=None):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/lucaskjaero/foreign-language-reader/pullRequest?prid=5112709)